mmv1/third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -146,6 +151,14 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				},
 			},
 
+			"docker_registry": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "CONTAINER_REGISTRY",
+				ValidateFunc: validation.StringInSlice(allowedDockerRegistrySettings, true),
```

Comment:
Generally we avoid this kind of validation unless this set is truly never going to change, because it means that users can be artificially held back from using the latest features of an API because of being on an older version of the provider.

Reply:
I see, would you suggest a better validation option in terms of forward compatibility? I kept with an existing pattern. Also, for the GCF Gen1, it is very unlikely that we will have a new type of image registry in the observable future.

Reply:
Generally we just don't validate enums like this at all. If you're confident that it is unlikely to change, we can leave the validation in place for now. Basically - if it ever does end up being an issue we'll probably remove the validation at that time.

Reply:
Ok, I am sure this set won't be changed in the observable future, but I have removed the validation since that seems like an established practice here. Thanks!